flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveSource.java

```diff
@@ -46,29 +58,98 @@
 
 	private static final long serialVersionUID = 1L;
 
+	private final JobConfWrapper jobConfWrapper;
+	private final List<String> partitionKeys;
+	private final ContinuousPartitionFetcher<Partition, ?> fetcher;
+	private final HiveTableSource.HiveContinuousPartitionFetcherContext<?> fetcherContext;
+	private final ObjectPath tablePath;
+
 	HiveSource(
 			JobConf jobConf,
+			ObjectPath tablePath,
 			CatalogTable catalogTable,
 			List<HiveTablePartition> partitions,
 			@Nullable Long limit,
 			String hiveVersion,
 			boolean useMapRedReader,
-			boolean isStreamingSource,
+			@Nullable ContinuousEnumerationSettings continuousEnumerationSettings,
+			ContinuousPartitionFetcher<Partition, ?> fetcher,
+			HiveTableSource.HiveContinuousPartitionFetcherContext<?> fetcherContext,
 			RowType producedRowType) {
 		super(
 				new org.apache.flink.core.fs.Path[1],
 				new HiveSourceFileEnumerator.Provider(partitions, new JobConfWrapper(jobConf)),
-				DEFAULT_SPLIT_ASSIGNER,
+				continuousEnumerationSettings == null ? DEFAULT_SPLIT_ASSIGNER : SimpleSplitAssigner::new,
```

Comment:
Why not `DEFAULT_SPLIT_ASSIGNER `?

Reply:
In streaming read, we would want the splits to be consumed in order. The `DEFAULT_SPLIT_ASSIGNER` is locality-aware and therefore doesn't meet the requirement. But perhaps I should only do this for partitioned table, because non-partitioned table still reuse super class's enumerator and the splits are not generated in order in the first place.

Reply:
I think ideally we need to implement an assigner that assigns splits in order, but is also locality-aware. Guess we can leave that for the future.